pkg/oc/bootstrap/clusterup/componentinstall/apply_list.go

```diff
@@ -0,0 +1,90 @@
+package componentinstall
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/run"
+	"github.com/openshift/origin/pkg/oc/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type List struct {
+	ComponentName string
+
+	Image string
+
+	Namespace  string
+	KubeConfig []byte
+	List       []byte
+
+	WaitCondition func() (bool, error)
+}
+
+const listBashScript = `#!/bin/sh
+set -e
+set -x
+
+ls -alh /
+
+ns=""
+if [ -s /namespace-file ]; then
+	ns="--namespace=$(cat /namespace-file) "
+fi
+
+oc apply ${ns} --config=/kubeconfig.kubeconfig -f /list.yaml 
+`
+
+func (opt List) Name() string {
+	return opt.ComponentName
+}
+
+func (opt List) Install(dockerClient dockerhelper.Interface) error {
+	imageRunHelper := run.NewRunHelper(dockerhelper.NewHelper(dockerClient)).New()
+
+	glog.Infof("Installing %q", opt.Name())
+
+	contentToCopy := map[string][]byte{
+		"kubeconfig.kubeconfig": opt.KubeConfig,
+		"list.yaml":             opt.List,
+		"apply.sh":              []byte(listBashScript),
+		"namespace-file":        []byte(opt.Namespace),
+	}
+
+	var lastErr error
+	// do a very simple retry loop on failure. Three times, ten second gaps
+	wait.PollImmediate(10*time.Second, 30*time.Second, func() (bool, error) {
+		_, rc, err := imageRunHelper.Image(opt.Image).
+			Privileged().
+			DiscardContainer().
+			Copy(contentToCopy).
+			HostNetwork().
+			HostPid().
+			Entrypoint("sh").
+			Command("-c", "chmod 755 /apply.sh; /apply.sh").Run()
```

Comment:
nit: chmod 755 /apply.sh && /apply.sh (in case the file is missing, fail sooner)